harness/src/lib.rs

```diff
@@ -498,6 +501,35 @@ pub struct Mollusk {
     pub slot: u64,
 }
 
+pub trait InvocationInspectCallback {
+    fn before_invocation(
+        &self,
+        program_id: &Pubkey,
+        instruction_data: &[u8],
+        instruction_accounts: &[InstructionAccount],
+        program: IndexOfAccount,
+        invoke_context: &InvokeContext,
+    );
+
+    fn after_invocation(&self, invoke_context: &InvokeContext);
+}
+
+pub struct EmptyInvocationInspectCallback {}
```

Comment:
Does this need to be `pub`?

```suggestion
struct EmptyInvocationInspectCallback;
```